nomad/node_endpoint.go

```diff
@@ -835,26 +835,27 @@ func (n *Node) UpdateAlloc(args *structs.AllocUpdateRequest, reply *structs.Gene
 
 		// Add an evaluation if this is a failed alloc that is eligible for rescheduling
 		if alloc.ClientStatus == structs.AllocClientStatusFailed {
-			ws := memdb.NewWatchSet()
-			job, err := n.srv.State().JobByID(ws, alloc.Namespace, alloc.JobID)
-			if err != nil {
-				n.srv.logger.Printf("[ERR] nomad.client: Unable to find jobid %v", alloc.JobID)
-				return err
-			}
-			if job == nil {
-				return fmt.Errorf("[ERR] nomad.client: Unable to find jobid %v", alloc.JobID)
-			}
 			// Only create evaluations if this is an existing alloc,
 			// and eligible as per its task group's ReschedulePolicy
-			if existingAlloc, _ := n.srv.State().AllocByID(ws, alloc.ID); existingAlloc != nil {
+			if existingAlloc, _ := n.srv.State().AllocByID(nil, alloc.ID); existingAlloc != nil {
+				job, err := n.srv.State().JobByID(nil, existingAlloc.Namespace, existingAlloc.JobID)
+				if err != nil {
+					n.srv.logger.Printf("[WARN] nomad.client: UpdateAlloc unable to find job ID %q :%v", existingAlloc.JobID, err)
```

Comment:
This should be a log at ERR level. I think we can drop the job == nil one to debug